src/stateContext.js

```diff
@@ -0,0 +1,27 @@
+import { useReducer } from "react";
+import { createContext, useContext } from "react";
+import { ProductsDatabase } from "./ProductsPage";
+import { reducer } from "./stateReducer";
+export const ReducerContext = createContext();
+
+export const ReducerProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, {
```

Comment:
instead of naming this state, you can name is something better for readability such as user&ProductsState